packages/block-editor/src/store/selectors.js

```diff
@@ -1496,11 +1497,22 @@ const buildBlockTypeItem = ( state, { buildScope = 'inserter' } ) => (
 	const id = blockType.name;
 
 	let isDisabled = false;
+	let disabledMessage = '';
 	if ( ! hasBlockSupport( blockType.name, 'multiple', true ) ) {
+		// If the block doesn't support multiple instances, check the state for already added
+		// blocks of the same type, if it finds any, this block type becomes disabled
 		isDisabled = some(
 			getBlocksByClientId( state, getClientIdsWithDescendants( state ) ),
 			{ name: blockType.name }
 		);
+
+		if ( isDisabled ) {
```

Comment:
I think it would be better if the `disabledMessage` is created within the target `MenuItem` Component. `selectors` are not concerned with translations since that is related to the `View` in question. Let me know what you think.

Reply:
I see. My concern for adding it to the `selector` instead of the `MenuItem` was that we'd be adding a 'hidden' logic to the `MenuItem`. It knows that the button is disabled, but it doesn't know the reason, if we add it there, it would 'assume' it's because of the multiple instances of that block. And also, if for some reason a new reason for a button to be disabled appears, the message displayed would be wrong.

What do you think?

Reply:
I also see that some tests are breaking because of this, they're expecting an object from the `selector` and the object now contains an extra field. I'll wait to fix those after we make a decision on this :)

Reply:
We can pass something to identify that the block is already added to the editor at that moment, then in the `MenuItem` we can check if `isDisabled` is true and also if it's already added once, to then display the message. I believe we still have some hidden logic of associating the `isDisabled` with this new property. The added benefit is that translation would be handled by the `MenuItem`. But we'd still be modifying the `selectors.js`. 

If we handle all of this disabled logic in the `selectors.js`, in the future, Web could also display a similar message and it would be extensible in case we have more reasons for a button to be disabled.

Which way would be preferable? 😃

Reply:
Thanks for expounding here. Seems good to me to pass a new property so that the web could handle this logic in the future. 👍🏾

Reply:
As always there are multiple ways to resolve an issue. Are you seeing another location that handles translations within a store?

Reply:
It makes sense :) 

I'll add a property, but handle the translation in the `MenuItem`! Thanks for the guidance here!

Reply:
> As always there are multiple ways to resolve an issue. Are you seeing another location that handles translations within a store?

You're right, I see only one other place, which is the 'Publicize' plugin.

Reply:
Done!

https://github.com/WordPress/gutenberg/pull/35021/commits/ecf2f13bdf7214600e4e596d14ad6e8337f7eaea